spec.bs

```diff
@@ -1187,6 +1227,90 @@ To <dfn>report win</dfn> given a [=leading bid info=] |leadingBidInfo|, a [=stri
   1. TODO: Store |resultUrl| and |reportingBeaconMap| in the {{FencedFrameConfig}} as appropriate.
 </div>
 
+# K-anonymity # {#k-anonymity}
+  
+*This introduction sub-section is non-normative*.
+
+Two goals of this specification rely on applying k-anonymity thresholds:
+  
+  * To prevent cross-site leaks:  Inputs to event-level reporting functions, `reportWin()` and
+    `reportResult()`, only contain limited cross-site information. Part of this limiting is done by
+    applying k-anonymity requirements to the ad URLs.  See
+    [privacy considerations](#privacy-considerations).
+  * To prevent microtargeting:  The browser applies k-anonymity requirements on the ad URLs to
+    ensure that the same ad or ad component is being shown to at least some minimum number of
+    people.
+
+The browser enforces these k-anonymity requirements by maintaining counts of how many times certain
+ads have been shown to the user. Since these counts are maintained across users, the counting must
+be done on a browser-maintained server. This specification relies on two operations to query and
+increment the counts: [=query k-anonymity count=] and [=increment k-anonymity count=].
+  
+The details of how the browser-maintained server is operated and accessed are implementation details
+but it should be done in a way that prevents the server operator from joining the identity of two
+query or increment requests. One way to help prevent this is by making accesses to the server go
+through an HTTP proxy that prevents the server from seeing the browsers' IP addresses.
+
+The browser should choose a k-anonymity threshold, otherwise known as the value for "k", depending
+on the projected sizes of interest groups and their privacy goals. For example an implementation
+might choose to require a k-anonymity threshold of fifty users over a seven day period. The server
+will maintain the count over the chosen duration and compare the count to the chosen k-anonymity
+threshold when responding to [=query k-anonymity count=].
+
+<div algorithm>
+  To <dfn>query k-anonymity count</dfn> given a |hashCode|:
+    1. Query k-anonymity server for whether |hashCode|'s count is above the k-anonymity threshold.
+    1. Return true if it is above the threshold, otherwise return false.
+</div>
+
+<div algorithm>
+  To <dfn>query ad k-anonymity count</dfn> given an [=interest group=] |ig| and a [=URL=] |ad|:
+    1. Let |keyString| be the [=string=] concatenating "AdBid\n" and |ig|'s [=interest group/owner=]
+       and "\n" and |ig|'s [=interest group/bidding url=] and "\n" and |ad|.
+    1. Let |keyHash| be the [=SHA-256=] hash of |keyString|.
+    1. Return the result of [=query k-anonymity count|querying the k-anonymity count=] given |keyHash|.
+</div>
+
+<div algorithm>
+  To <dfn>query component ad k-anonymity count</dfn> given a [=URL=] |ad|:
+    1. Let |keyString| be the [=string=] concatenating "ComponentBid\n" and |ad|.

```

Comment:
C escapes aren't part of the string literal syntax (https://infra.spec.whatwg.org/#strings), so this says to use the two characters `\` and `n` to separate the components. Instead, you can write:

```suggestion
    1. Let |keyString| be the [=string/concatenation=] of the following strings separated with U+000A LF:
        1. "ComponentBid" 
        1. the [=URL serializer|serialization=] of |ad|.
```

And similarly for the other places that use "\n".